subprojects/scala/src/main/java/org/gradle/api/plugins/scala/ScalaBasePlugin.java

```diff
@@ -104,6 +112,9 @@ public void apply(final Project project) {
     private void configureConfigurations(final Project project, final Usage incrementalAnalysisUsage, ScalaPluginExtension scalaPluginExtension) {
         DependencyHandler dependencyHandler = project.getDependencies();
 
+        Configuration plugins = project.getConfigurations().create(SCALA_COMPILER_PLUGINS_CONFIGURATION_NAME);
+        plugins.setTransitive(false);
```

Comment:
Please set the `canBeConsumed` flag to `false`

Reply:
In addition, the configuration needs to have attributes for resolution. It should have the same attributes as a runtime classpath. So you can check which attributes are defined in the `JavaBasePlugin` class for the `runtimeClasspath` configuration - around lines 321 to 325.

Reply:
Added requested attributes